android-agent/src/main/kotlin/io/opentelemetry/android/agent/session/factory/SessionManagerFactory.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.agent.session.factory
+
+import android.app.Application
+import io.opentelemetry.android.agent.session.SessionConfig
+import io.opentelemetry.android.agent.session.SessionIdTimeoutHandler
+import io.opentelemetry.android.agent.session.SessionManager
+import io.opentelemetry.android.internal.services.Services
+import io.opentelemetry.android.session.SessionProvider
+
+/**
+ * Default implementation of [SessionProviderFactory] that creates [SessionManager] instances.
+ *
+ * This implementation creates a [SessionManager] and wires it with:
+ * - A timeout handler configured from the session config
+ * - Application lifecycle integration for timeout management
+ *
+ * This class is open to allow custom implementations for specialized use cases.
+ *
+ * @see SessionProviderFactory
+ * @see SessionManager
+ * @see SessionConfig
+ */
+open class SessionManagerFactory : SessionProviderFactory {
```

Comment:
I'd suggest making this `internal` so that it isn't exposed, as I'd consider this an implementation detail. As long as the `SessionProviderFactory` type is public and there's a way to set it on the `OpenTelemetryConfiguration` DSL, that covers the use-case of providing custom implementations.